Kernel/Devices/Storage/StorageManagement.cpp

```diff
@@ -205,21 +211,26 @@ UNMAP_AFTER_INIT ErrorOr<NonnullOwnPtr<Partition::PartitionTable>> StorageManage
     return TRY(Partition::GUIDPartitionTable::try_to_initialize(device));
 }
 
+void StorageManagement::enumerate_device_partitions(StorageDevice& device)
+{
+    auto partition_table_or_error = try_to_initialize_partition_table(device);
+    if (partition_table_or_error.is_error())
+        return;
+    auto partition_table = partition_table_or_error.release_value();
+    for (size_t partition_index = 0; partition_index < partition_table->partitions_count(); partition_index++) {
+        auto partition_metadata = partition_table->partition(partition_index);
+        if (!partition_metadata.has_value())
+            return;
```

Comment:
I don't think you need `partition_index` here, you can just iterate over `partition_table->partitions()`?

Reply:
That made the function quite a bit smaller actually